chain/chain/src/chain.rs

```diff
@@ -4771,27 +4771,23 @@ impl Chain {
         receipts: &[Receipt],
         shard_layout: &ShardLayout,
     ) -> Vec<CryptoHash> {
-        if shard_layout.num_shards() == 1 {
+        let num_shards = shard_layout.shard_ids().count();
+        if num_shards == 1 {
```

Comment:
I am tempted to get rid of this optimisation.  In practice, we will never hit this case.